### DISCUSSION TITLE
Setting unique titles for xarray facetgrid

### DISCUSSION THREAD

**Discussion by Murk89** [pydata/xarray#5823] [Q&A] [answered]
Hi,
I am plotting a temperature variable T2 from my climate model WRF. I am able to plot an xarray facetgrid from the output. What I am struggling with is the titles of the subplots. How to change the title of each plot to give it a title of my choice? Currently the titles of each subplot, i.e. month=1, and so on, are automatically generated as I have stated row='month'. I am not clear how to use the ax.set_title arguments for each subplot, when they are part of the facetgrid. Below is my code, and the current facetgrid plot. Would really appreciate some help. 

from __future__ import print_function
import xarray as xr
import pandas as pd
import matplotlib.pyplot as plt`

import netCDF4
from netCDF4 import Dataset
import matplotlib.pyplot as plt
from matplotlib.cm import get_cmap
import cartopy
import cartopy.crs as ccrs
import cartopy.feature as cf
from cartopy.feature import NaturalEarthFeature

from netCDF4 import num2date
import numpy as np
import os
import pandas as pd

from netCDF4 import Dataset
from wrf import getvar, interpline, CoordPair, xy_to_ll, ll_to_xy
from wrf import get_cartopy, geo_bounds
#import DataArray.plot.surface
import seaborn as sns
import matplotlib.dates as mdates
from matplotlib.dates import DateFormatter
from wrf import (to_np, getvar, smooth2d, get_cartopy, cartopy_xlim,cartopy_ylim, latlon_coords)

dataset=xr.open_mfdataset("Python files for plotting wrfoutput\era5_1990-2000_output\*.nc",concat_dim='Time', combine='nested', compat='no_conflicts', preprocess=None, engine=None, data_vars='all', coords='all', parallel=False, join='outer', attrs_file=None,)

DS = dataset.rename({'XTIME':'Time'})

#Convert to datetime. 
DS = DS.assign_coords(Time=pd.to_datetime(DS['Time'].values))

#convert to celsius and drop na
C = DS.T2.dropna(dim='Time') - 273.15

B=C.groupby('Time.month').mean()

#define levels for the colourbar
levels = [-15, -14, -13,-12,-11,-10,-9,-8,-7,-6,-5,-4,-3,-2,-1, 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]
g=B.plot(row='month', col_wrap=2, cmap='RdBu', levels=levels, extend='both',cbar_kwargs={'label': 'T2 (°C)'})

![Figure_1](https://user-images.githubusercontent.com/59877746/134787386-a10b47c9-5954-4d62-8c38-3f229ea5c751.png)

**Comment by spencerkclark** [comment, marked as answer]
If you'd like to have control over the axes directly, you can access them using the `axes` attribute of the returned [`FacetGrid`](http://xarray.pydata.org/en/stable/generated/xarray.plot.FacetGrid.html#xarray-plot-facetgrid) object.  For example:
```python
for ax, title in zip(g.axes.flat, list("abcd")):
    ax.set_title(title)
```
[This part of the documentation](http://xarray.pydata.org/en/stable/user-guide/plotting.html#facetgrid-objects) covers this and a few other useful features of the `FacetGrid` object.

**Reply by Murk89** [reply]
Hi spencer,
Many thanks for the help. So your code names the subplots a b c d individually. What if I have to name them e.g. 1990, 1991, 1992 and 1993 or hot year, mild year, cold year, coldest year. I understand that the zip argument has to use an iteration? Not sure how to go about it TBH. 
Thanks for sharing the documentation, I have tried to apply their suggestions as well, but somehow I still cannot figure out how to name them individually; since even in this documentation there is an iteration with the same initial title 'Air temperature'. and I want unique titles for each subplot.

**Reply by spencerkclark** [reply]
Yes, `zip` is commonly used in iterating over multiple lists simultaneously.  To title the axes with other values, you can define your own list to iterate over:
```python
titles = ["hot year", "mild year", "cold year", "coldest year"]
for ax, title in zip(g.axes.flat, titles):
    ax.set_title(title)
```
If you'd like to do things without iteration, you can index into the array of axes of the `FacetGrid` object.  For example to set the title of just the first panel you can use:
```python
g.axes[0, 0].set_title("Title of the first panel")
```

**Reply by Murk89** [reply]
Thankyou so much for explaining things so nicely. It works a charm! :-)
